sound/soc/sof/loader.c

```diff
@@ -20,13 +20,21 @@ static int get_ext_windows(struct snd_sof_dev *sdev,
 {
 	const struct sof_ipc_window *w =
 		container_of(ext_hdr, struct sof_ipc_window, ext_hdr);
+	size_t w_size = struct_size(w, window, w->num_windows);
+
+	if (sdev->info_window) {
+		if (memcmp(sdev->info_window, w, w_size)) {
```

Comment:
you compare `w_size` of memory here, and `w_size` is calculated from `w->num_windows`, and `w->num_windows` is only verified below to be valid. So, for example if `w->num_windows == 0` the below check would return `-EINVAL`, but in case of `sdev->info_window != NULL` you now return 0 in this case? Which one is correct?

Reply:
Such an issue would be properly handled, because for `w->num_windows == 0` will differ from `sdev->info_window->num_windows` for sure (condition in line 24/33). Event that I changed order of those checks, to make it more obvious.